File: 001_xxDevPlus/xxDevPlus/ObjPile.cs
//** @author  GeTr Copyright© 1996-2020 Gerald Trost, Klagenfurt, AUSTRIA. No warranties or liabilities whatsoever, not with third parties
//** @license You may use, modify, distribute this code under GNU public dual licence free for NON-Commercial use as long as this notice remains unchanged.
//** @comment Pile<Object> for convenience




using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace org_xxdevplus_struct

{
 public class ObjPile : Pile<object>
 {
  //Copyright© 1996-2012 Gerald Trost, Klagenfurt, AUSTRIA. No warranties or liabilities whatsoever, not with third parties, You may use, modify, distribute this code under GNU public licence for NON-Commercial use as long as this notice remains unchanged.
  private static bool selfTested  = false;  private static string ass(bool expr) { if (!expr) throw new Exception(ass(true) + " SelfTest Failure"); return "DatPile"; } private static void selfTest() { selfTested = true; } private void init() { if (!selfTested) selfTest(); }
 
  public  ObjPile(params object[] obj) : base("", true, obj) { }

 }
}

File: 001_xxDevPlus/src/org/xxdevplus/struct/ObjPile.java

//** @author  GeTr Copyright© 1996-2020 Gerald Trost, Klagenfurt, AUSTRIA. No warranties or liabilities whatsoever, not with third parties
//** @license You may use, modify, distribute this code under GNU public dual licence free for NON-Commercial use as long as this notice remains unchanged.
//** @comment Pile<Object> for convenience

package org.xxdevplus.struct;

import org.xxdevplus.struct.Pile;

 
 public class ObjPile extends Pile<Object>
 {
  private static boolean selfTested  = false;  private static String ass(boolean expr) throws Exception { if (!expr) throw new Exception(ass(true) + " SelfTest Failure"); return "DatPile"; } 
  private static void selfTest() { selfTested = true; }
 
  public  ObjPile(Object... obj) { super("", 0, obj); }

 }
